metrics/local_test.go

```diff
@@ -10,11 +9,6 @@ import (
 )
 
 func TestLocalMetrics(t *testing.T) {
-	numGoroutines := runtime.NumGoroutine()
-	defer func() {
-		assert.Equal(t, numGoroutines, runtime.NumGoroutine(), "Leaked at least one goroutine.")
-	}()
```

Comment:
my guess is the deferred localbackend.stop() and this defer func have a race condition. Ofc the localbackend.stop() will run first but the check that stops the actual go routine doesn't die before this function gets called. I can't reproduce locally so it's just a guess.